cli/local/api.go

```diff
@@ -27,17 +27,40 @@ import (
 	"github.com/cortexlabs/cortex/pkg/lib/files"
 	"github.com/cortexlabs/cortex/pkg/lib/msgpack"
 	"github.com/cortexlabs/cortex/pkg/lib/pointer"
+	"github.com/cortexlabs/cortex/pkg/lib/prompt"
 	"github.com/cortexlabs/cortex/pkg/lib/sets/strset"
 	"github.com/cortexlabs/cortex/pkg/types/spec"
 	"github.com/cortexlabs/cortex/pkg/types/userconfig"
 )
 
 var _deploymentID = "local"
 
-func UpdateAPI(apiConfig *userconfig.API, configPath string, projectID string, awsClient *aws.Client) (*spec.API, string, error) {
+func UpdateAPI(apiConfig *userconfig.API, configPath string, projectID string, deployDisallowPrompt bool, awsClient *aws.Client) (*spec.API, string, error) {
+	var incompatibleVersion string
+	encounteredVersionMismatch := false
 	prevAPISpec, err := FindAPISpec(apiConfig.Name)
 	if err != nil {
-		if errors.GetKind(err) != ErrAPINotDeployed {
+		if errors.GetKind(err) == ErrCortexVersionMismatch {
```

Comment:
I believe `ErrCortexVersionMismatch` will be raised if the Cortex versions don't match exactly. Let's say you've deployed an API with 0.18.0 and you upgraded your CLI to 0.18.1. You try to redeploy but you run into a version mismatch:

```
api iris-classifier was deployed using CLI version 0.18.0 but the current CLI version is 0.18.1; re-deploying iris-classifier with current CLI version 0.18.1 might yield an unexpected outcome; any cached models won't be deleted

it is recommended to download version 0.18.0 of the CLI from https://docs.cortex.dev/v/0.18/install, delete the API using version 0.18.0 of the CLI and then re-deploy the API using the latest version of the CLI

do you still want to re-deploy?
```

If you install the CLI docs by following `https://docs.cortex.dev/v/0.18/install`, you will still download 0.18.1 CLI because you will download the latest patch version of that minor version. You would run into the same prompt again. Maybe this prompt should only be shown if the minor versions don't match. In a scenario where minor versions do match, I think we can assume that it is safe to re-deploy or delete. Let me know what you think.

Reply:
Agreed. The prompt should only be shown when minor versions don't match. Patches shouldn't change the functionality, but just add fixes.